pkg/rtc/mediatrack.go

```diff
@@ -469,6 +473,30 @@ func (t *MediaTrack) RemoveAllSubscribers() {
 	t.subscribedTracks = sync.Map{}
 }
 
+func (t *MediaTrack) RevokeDisallowedSubscribers(allowedSubscriberIDs []string) []string {
+	t.lock.Lock()
+	defer t.lock.Unlock()
+
+	var revokedSubscriberIDs []string
+	t.subscribedTracks.Range(func(key interface{}, val interface{}) bool {
```

Comment:
with sync.Map, maybe not need lock here.

what about this loop:
```
for _, disallowdId := range allowedSubscriberIDs {
     if subTrack,ok := t.subscribedTracks.Load(disallowedId);ok {
           ....
     }
}
}
```

Reply:
Good idea @cnderrauber . Stupid of me not to do that in the first place :-) Thank you 🙇

Reply:
Actually, my bad. The allowed list is going to be small usually. The loop that is long is currently active subscriptions. Let's say that there are 100K subscribers on a track. Then the publisher decides that they want to allow only two subscribers. We have to shutdown 99998 subscriptions. So, have to loop through the current active subscriptions.

Reply:
sorry, I'm confused by the name and your comment. If allowd list is small, why we have to shutdown 99998? do you going to close the subscriber that not in allow list? but in code, looks you are close subscribe in allow list

Reply:
and if there are 99998 subscriber to be closed, I think better to use a single goroutine to do that, not 99998 goroutines, if it will not cause a timing issue.

Reply:
Sorry, my bad @cnderrauber . That is a bug in the code. Everything not in the allow list need to be closed. Will fix it now. Will think about the single goroutine option later. We have a bunch of these loops we need to take care of. Probably need a full sweep with a 100K person room in mind at some point.

Reply:
Thanks for catching it :-)